imaspy/test/test_nc_autofill.py

```diff
@@ -1,11 +1,56 @@
 from imaspy.db_entry import DBEntry
+from imaspy.exception import InvalidNetCDFEntry
 from imaspy.test.test_helpers import compare_children, fill_consistent
+import re
+import pytest
+import netCDF4
+from packaging import version
 
 
-def test_nc_latest_dd_autofill_put_get(ids_name, tmp_path):
+def test_nc_latest_dd_autofill_put_get_skip_complex(ids_name, tmp_path):
     with DBEntry(f"{tmp_path}/test-{ids_name}.nc", "x") as entry:
         ids = entry.factory.new(ids_name)
-        fill_consistent(ids, 0.5)
+        fill_consistent(ids, leave_empty=0.5, skip_complex=True)
+
+        entry.put(ids)
+        ids2 = entry.get(ids_name)
+
+    compare_children(ids, ids2)
+
+
+@pytest.mark.skipif(
+    version.parse(netCDF4.__version__) < version.parse("1.7.0"),
+    reason="NetCDF4 versions < 1.7.0 do not support complex numbers",
+)
+def test_nc_latest_dd_autofill_put_get_with_complex_older_netCDF4(
+    ids_name, tmp_path
+):
+    with DBEntry(f"{tmp_path}/test-{ids_name}.nc", "x") as entry:
+        ids = entry.factory.new(ids_name)
+        fill_consistent(ids, leave_empty=0.5, skip_complex=False)
+        try:
+            entry.put(ids)
+            ids2 = entry.get(ids_name)
+            compare_children(ids, ids2)
+        except InvalidNetCDFEntry as e:
```

Comment:
This test is skipped when the netCDF library is too old, so this exception should never trigger?

If you agree, please remove the try/except and keep the test simpler :)

Reply:
oh dear, this is a bug in the test skipIf logic I added. I've updated so it should now be correct.